conans/test/command/package_id_test.py

```diff
@@ -0,0 +1,72 @@
+import unittest
+import os, re
+from conans.test.tools import TestClient
+from conans.test.utils.test_files import temp_folder
+from conans.paths import CONANFILE
+import platform
+
+short_path_file = """
+from conans import ConanFile
+
+class AConan(ConanFile):
+    name = "MyPackage"
+    version = "0.1.0"
+    short_paths=True
+"""
+
+
+class PackageIdTest(unittest.TestCase):
+    def setUp(self):
+        self.settingsA = ("-s os=Windows -s compiler='Visual Studio' -s compiler.version=12 "
+                         "-s arch=x86 -s compiler.runtime=MD")
+
+        self.settingsB = ("-s os=Windows -s compiler='Visual Studio' -s compiler.version=14 "
+                          "-s arch=x86 -s compiler.runtime=MDd")
+
+    def test_basic(self):
+        client = TestClient()
+        client.run("new MyPackage/0.1.0@myuser/testing")
+        for settings in [self.settingsA, self.settingsB]:
+            client.run("path_info myUser %s" % settings)
+            basePath = os.path.join("MyPackage", "0.1.0", "myUser", "testing");
+            output = client.user_io.out;
+            self.assertIn(os.path.join(basePath, "export"), output)
+            self.assertIn(os.path.join(basePath, "source"), output)
+
+            client.run("info %s" % settings)
+            id = re.search('ID:\s*([a-z0-9]*)', str(client.user_io.out)).group(1);
+
+            self.assertIn(os.path.join(basePath, "build", id), output)
+            self.assertIn(os.path.join(basePath, "package", id), output)
+
+    def test_short_pathes(self):
+        if platform.system() == "Windows":
+            currentEnv = os.environ.get("CONAN_USER_HOME_SHORT", None)
```

Comment:
Use `tools.environment_append({"CONAN_USER_HOME_SHORT": short_folder})` context manager to avoid finally